hack/crdbversions/main.go

```diff
@@ -0,0 +1,164 @@
+	{"config/templates/crdb-tls-example.yaml.in", "config/samples/crdb-tls-example.yaml"},
+	{"config/templates/example.yaml.in", "examples/example.yaml"},
+}
+
+// crdb-versions.yaml structure
+type crdbVersions struct {
+	CrdbVersions []string `yaml:"CrdbVersions"`
+}
+
+type templateData struct {
+	CrdbVersions            []*semver.Version
+	LatestStableCrdbVersion string
+	OperatorVersion         string
+	GeneratedWarning        string
+}
+
+// readCrdbVersions reads CRDB versions from a YAML file and sorts them
+// according to the semantic version sorting rules
+func readCrdbVersions(r io.Reader) ([]*semver.Version, error) {
+	contents, err := io.ReadAll(r)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open CRDB version file: %w", err)
+	}
+	var versions crdbVersions
+	if err := yaml.Unmarshal(contents, &versions); err != nil {
+		return nil, fmt.Errorf("cannot parse CRDB version file: %w", err)
+	}
+	var vs []*semver.Version
+	for _, raw := range versions.CrdbVersions {
+		v, err := semver.NewVersion(raw)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert version `%s`: %w", r, err)
+		}
+		vs = append(vs, v)
+	}
+	sort.Sort(semver.Collection(vs))
+	return vs, nil
+}
+
+func generateTemplateData(crdbVersions []*semver.Version, operatorVersion string) (templateData, error) {
+	var data templateData
+	data.OperatorVersion = operatorVersion
+	data.GeneratedWarning = "Generated, do not edit"
+	data.CrdbVersions = crdbVersions
+	var stableVersions []*semver.Version
+	for _, v := range data.CrdbVersions {
+		if isStable(*v) {
+			stableVersions = append(stableVersions, v)
+		}
+	}
+	if len(stableVersions) == 0 {
+		return templateData{}, fmt.Errorf("cannot find stable versions")
+	}
+	latestStable := stableVersions[len(stableVersions)-1]
+	data.LatestStableCrdbVersion = latestStable.Original()
+	return data, nil
+}
+
+func isStable(v semver.Version) bool {
+	return v.Prerelease() == "" && v.Metadata() == ""
+}
+
+func dotsToUnderscore(v semver.Version) string {
+	return strings.ReplaceAll(v.Original(), ".", "_")
+}
+
+func generateFile(name string, tplText string, output io.Writer, data templateData) error {
+	// Teamplate functions
+	funcs := template.FuncMap{
+		"underscore": dotsToUnderscore,
+		"stable":     isStable,
+	}
+	tpl, err := template.New(name).Funcs(funcs).Parse(tplText)
+	if err != nil {
+		return fmt.Errorf("cannot parse `%s`: %w", name, err)
+	}
+	return tpl.Execute(output, data)
+}
+
+func main() {
+	log.SetFlags(0)
+	crdbVersionsFile := flag.String("crdb-versions", "", "YAML file with CRDB versions")
+	operatorVersion := flag.String("operator-version", "", "Current operator version")
+	repoRoot := flag.String("repo-root", "", "Git repository root")
+	flag.Parse()
+
+	if *crdbVersionsFile == "" || *operatorVersion == "" || *repoRoot == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	f, err := os.Open(*crdbVersionsFile)
+	if err != nil {
+		log.Fatalf("Cannot open versions file: %s", err)
```

Comment:
does Fatalf panic?

Reply:
It cals `os.Exit()`.  Technically it's not the same as `panic`, but at least the logic stops there.